src/Sarif/Extensions.cs

```diff
@@ -235,8 +235,11 @@ public static Uri CreateUriForJsonSerialization(this string uriText)
             // Uri, so we construct a Uri first and then, if necessary, 
             // create a second one using the file:/// representation as
             // the original string passed to the constructor instance.
-            Uri result = new Uri(uriText, UriKind.RelativeOrAbsolute);
-            if (result.IsFile && !uriText.StartsWith("file://"))
+            Uri result;
+
+            if (Uri.TryCreate(uriText, UriKind.RelativeOrAbsolute, out result) &&
+                result.IsFile &&
+                result.Scheme != "file")
             {
                 result = new Uri(result.ToString());
             }
```

Comment:
Nothing. :) null return is ok is if uri can't be parsed, and there's no additional work to do if result isn't a file uri or if the uri scheme is already correct.